src/components/advanced-search/advanced-search-dates.vue

```diff
@@ -0,0 +1,139 @@
+<template>
+  <v-dialog id="advanced-search-dates" v-model="dialog" persistent max-width="44rem">
+    <v-card height="32rem" class="py-6">
+      <v-row>
+        <v-col>
+          <span class="title-bold-16" :class="{ 'error-message': validate && !hasStartDate }">Select Start Date:</span>
+          <v-date-picker
+            id="start-date-calendar"
+            class="mt-2"
+            v-model="startDateText"
+            :max="currentDateString"
+            elevation="5"
+          >
+          </v-date-picker>
+        </v-col>
+        <v-col>
+          <span class="title-bold-16" :class="{ 'error-message': validate && !hasEndDate }">Select Start Date:</span>
+          <v-date-picker
+            id="end-date-calendar"
+            class="mt-2"
+            v-model="endDateText"
+            :max="currentDateString"
+            elevation="5"
+          >
+          </v-date-picker>
+        </v-col>
+      </v-row>
+      <v-card-actions class="my-0 py-0 float-right">
+          <v-btn id="btn-done" text color="primary" :ripple="false" @click="submitDates()"><strong>OK</strong></v-btn>
+          <v-btn id="btn-cancel" text color="primary" :ripple="false" @click="emitClose()">Cancel</v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang='ts'>
+import { Component, Emit, Prop, Mixins } from 'vue-property-decorator'
+import { Getter } from 'vuex-class'
+
+// Mixins
+import { DateMixin } from '@/mixins'
+
+// Interfaces
+import { StartEndDatesI } from '@/interfaces/models'
+
+@Component({})
+export default class AdvancedSearchDates extends Mixins(DateMixin) {
+  /** Prop to display the dialog. */
+  @Prop() private dialog: boolean
```

Comment:
Change private to readonly.

Reply:
kk, will do 👍

Reply:
Do you think all the props should be readonly?

Reply:
Great points!

Absolutely. I think the really valuable ones are when you don't close dialogs correctly. As an example you open another dialog without actually closing the initial one, you get this same error.